ddtrace/debugging/probe/model.py

```diff
@@ -0,0 +1,77 @@
+import abc
+from os.path import normcase
+from os.path import normpath
+from types import CodeType
+from typing import Optional
+
+import attr
+import six
+
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.rate_limiter import BudgetRateLimiterWithJitter as RateLimiter
+
+
+log = get_logger(__name__)
+
+
+def _normalize_path(path):
+    return normpath(normcase(path))
+
+
+@attr.s(hash=True)
```

Comment:
Don't have a ton of background but looking here it seems you should only set hash if you know exactly what you're doing?... Do you know exactly what you're doing? https://www.attrs.org/en/stable/hashing.html#hash-method-generation

Reply:
If I understood the documentation correctly, I _think_ I know what I'm doing 🙂 . A probe object should be hashable but not immutable (so it can't be frozen). Each probe is identified by a UUID, so for the hash function we can take the hash of the string representing the UUID, and for equality we rely on string comparison. This should allow us to put probe objects into hashable collections, like `dict`s and `set`s.

Reply:
This makes sense, thanks for the clarification! Looking forward to future PRs about the debugger so I can better understand the full scope of the project, but this PR itself looks good.